clients/src/main/java/org/apache/kafka/clients/ClientRequest.java

```diff
@@ -126,7 +115,8 @@ public long sendTimeMs() {
         return sendTimeMs;
     }
 
-    public void setSendTimeMs(long sendTimeMs) {
+    public void setSend(Send send, long sendTimeMs) {
```

Comment:
The challenge is that it's really helpful to have access to the `AbstractRequest` instance within `MockClient` for testing, so losing that seems like a step backwards. Take a look at the most recent commit and let me know what you think. I've created a separate object to represent the state of an in-flight request. I think this provides a little better encapsulation of the some of `NetworkClient` internals, such as working with the `Send` instance and distinguishing requests generated by the `NetworkClient` itself.